bin/split_fasta.py

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env python
+
+#USAGE: ./combine_tables.py <*.unbinned.fa> <length threshold> <maximal number of sequences> <length threshold to retain contigs>
+
+import pandas as pd
+from sys import argv
+from Bio import SeqIO
+from Bio.Seq import Seq
+from Bio.SeqRecord import SeqRecord
+from Bio.Alphabet import generic_dna
+import os
+
+# Input
+input_file = argv[1]
+length_threshold = int(argv[2])
+max_sequences = int(argv[3])
+min_length_to_retain_contig = int(argv[4])
+
+# Base name for file output
+out_base = (os.path.splitext(input_file)[0])
+
+# Read file
+fasta_sequences = SeqIO.parse(open(input_file),'fasta')
```

Comment:
files should be opened with the `with` context manager. that way the file handles will close automatically at the end of the scope.

Reply:
i.e.
```python
with open(input_file) as f:
    fasta_sequences = SeqIO.parse(f, 'fasta')
    # rest of the code that needs the file to be open.
```

Reply:
Right! Changed in https://github.com/nf-core/mag/pull/29/commits/8effb0b97c6d894e046f28b6c243df751e39a261